source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -413,6 +413,14 @@ class ComponentInterface : public NodeT {
    */
   rcl_interfaces::msg::SetParametersResult on_set_parameters_callback(const std::vector<rclcpp::Parameter>& parameters);
 
+  /**
+   * @brief Parameter validation wrapper that validates the period and calls the validation function of the derived
+   * Component classes.
+   * @param parameter A ParameterInterface pointer to a Parameter instance
+   * @return The validation result
+   */
+  virtual bool validate_parameter_wrapper(const std::shared_ptr<state_representation::ParameterInterface>& parameter);
```

Comment:
It's marked as `virtual`, but it is also `private`. I think you should remove `virtual`. 
```suggestion
  bool validate_parameter_wrapper(const std::shared_ptr<state_representation::ParameterInterface>& parameter);
```

Reply:
Copy error of course, thanks